torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1206,6 +1209,35 @@ def summon_full_params(
                     self._use_param_local_shard()
                     self.training_state = TrainingState_.IDLE
 
+    def named_parameters(
+        self,
+        prefix: str = "",
+        recurse: bool = True,
+    ) -> Iterator[Tuple[str, torch.nn.Parameter]]:
+        """
+        Overrides :meth:`torch.nn.Module.named_parameters()` to intercept
+        parameter names and remove all occurrences of the FSDP-specific
+        flattened parameter prefix when inside the :meth:`summon_full_params`
+        context manager.
+        """
+        # Determine which logic to use based on the context at call time
+        if not hasattr(self, "training_state") or \
```

Comment:
`nn.Modules` contained in an `FSDP` instance may not have `training_state` as an attribute but still return `True` for `isinstance(module, FullyShardedDataParallel)`. If you guys think this `hasattr()` check is too hacky, I will see if I can find a more direct solution.